src/test/java/io/quarkus/search/app/SearchServiceTest.java

```diff
@@ -118,41 +118,52 @@ public void relevance(String query, List<String> expectedGuideIds) {
 
     private static List<Arguments> relevance_params() {
         return List.of(
+                // I wonder if we could use something similar to https://stackoverflow.com/a/74737474/5043585
+                // to have some sort of weight in the documents and prioritize some of them
+                // problem will be to find the right balance because the weight would be always on
+                // another option could be to use the keyword to trick some searches
+                // I suspect we should make the keywords a List separated by , and a proper list in the index rather than what it is currently
```

Comment:
Not sure we can expect all documents to reliably use the proper syntax...

Regardless, in the full-text index you'll get the exact same result, whether you split the list yourself or you let the tokenizer do it. So unless you need spaces or punctutation in keywords, or you need faceting, I don't see what this would bring.